### PR TITLE
[REF] refactor doc strings for sample_mask and confounds for maskers

### PR DESCRIPTION
**Pull request by Remi-Gau**
- Closes none

Changes proposed in this pull request:

- use single doc strings for confounds and sample_mask for maskers methods
- add estimator checks 'tests' for maskers to transform iamges using sample_mask and confounds

### CODE REVIEW

File: nilearn/maskers/base_masker.py [hunk 1]
```diff
@@ -325,13 +312,9 @@ def fit_transform(
         y : numpy array of shape [n_samples], default=None
             Target values.
 
-        confounds : :obj:`list` of confounds, default=None
-            List of confounds (2D arrays or filenames pointing to CSV
-            files). Must be of same length than imgs_list.
```
Comment (hndgzkn) [inline nilearn/maskers/base_masker.py:330]: This line is not covered with common documentation.

**Review by hndgzkn — APPROVED** [review]
Minor comments, otherwise OK.

File: nilearn/_utils/docs.py [hunk 2]
```diff
@@ -807,9 +809,9 @@ def custom_function(vertices):
 
 # sample_mask
 docdict["sample_mask"] = """
-sample_mask : Any type compatible with numpy-array indexing, optional
+sample_mask : Any type compatible with numpy-array indexing, default=None
     shape: (number of scans - number of volumes removed, )
-    Masks the niimgs along time/fourth dimension to perform scrubbing
+    Masks the images along time dimension to perform scrubbing
```
[the remark this answers is not in the document]
Comment (Remi-Gau) [reply nilearn/_utils/docs.py:814]: > I think keeping `fourth` in parenthesis would be good.

yeah but we need a description that works for surface images and they have at most 2 dimensions

**Comment by Remi-Gau** [comment]
> Minor comments, otherwise OK.

I think I will add some common tests to all maskers to ensure that doc strings are well in line with behavior

File: nilearn/_utils/docs.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (hndgzkn) [reply nilearn/_utils/docs.py:814]: Is time dimension always the last dimension? May be that can be mentioned then.

File: nilearn/maskers/multi_nifti_masker.py [hunk 3]
```diff
@@ -499,14 +495,9 @@ def transform(self, imgs, confounds=None, sample_mask=None):
...
 
-        confounds : CSV file or 2D :obj:`numpy.ndarray` or \
-                :obj:`pandas.DataFrame`, default=None
-            This parameter is passed to signal.clean. Please see the
-            corresponding documentation for details.
+        %(confounds)s
 
-        sample_mask : :obj:`list` of 1D :obj:`numpy.ndarray`, default=None
-            List of sample_mask (1D arrays) if scrubbing motion outliers.
-            Must be of same length than imgs_list.
```
[the remark this answers is not in the document]
Comment (hndgzkn) [reply nilearn/maskers/multi_nifti_masker.py:509]: When I say this line, I mean `Must be of same length than imgs_list.`

File: nilearn/_utils/docs.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (Remi-Gau) [reply nilearn/_utils/docs.py:814]: actually saying the last dimension may be better indeed as it may also be other things that time

File: nilearn/_utils/docs.py [hunk 4]
```diff
@@ -254,7 +254,9 @@ def custom_function(vertices):
 
 # confounds
 docdict["confounds"] = """
-confounds : CSV file or array-like, optional
+confounds : :class:`numpy.ndarray`, :obj:`str`, :class:`pathlib.Path`, \
+            :class:`pandas.DataFrame` \
```
[the remark this answers is not in the document]
Comment (Remi-Gau) [reply nilearn/_utils/docs.py:258]: I deferred to the signal.clean doc string in most cases but added tests to check loading files from txt, csv, tsv...

File: nilearn/maskers/multi_nifti_masker.py [hunk 5]
```diff
@@ -441,9 +437,19 @@ def transform_imgs(
...
+                f"number of images ({len(imgs_list)})."
+            )
 
         if sample_mask is None:
             sample_mask = itertools.repeat(None, len(imgs_list))
+        elif len(sample_mask) != len(imgs_list):
+            raise ValueError(
+                f"number of sample_mask ({len(sample_mask)}) unequal to "
+                f"number of images ({len(imgs_list)})."
+            )
```
Comment (Remi-Gau) [inline nilearn/maskers/multi_nifti_masker.py:452]: the errors I was getting when passing the wrong number of confounds or masks was not really clear: hopefully this should help

**Review by bthirion — APPROVED** [review]
